packages/playground/remote/src/lib/offline-mode-cache.ts

```diff
@@ -0,0 +1,97 @@
+import { isURLScoped } from '@php-wasm/scopes';
+
+export class OfflineModeCache {
+	readonly cacheNamePrefix = 'playground-cache';
+
+	private cacheName: string;
+	private hostname: string;
+
+	constructor(cacheVersion: string, hostname: string) {
+		this.hostname = hostname;
+		this.cacheName = `${this.cacheNamePrefix}-${cacheVersion}`;
+	}
+
+	addCache = async (key: Request, response: Response) => {
+		const clonedResponse = response.clone();
+		const cache = await caches.open(this.cacheName);
+		await cache.put(key, clonedResponse);
+	};
+
+	getCache = async (key: Request) => {
+		const cache = caches.open(this.cacheName);
+		return await cache.then((c) => c.match(key, { ignoreSearch: true }));
+	};
+
+	shouldCacheUrl = (url: URL) => {
```

Comment:
This is one eager function. It will cache everything we request on the playground.wordpress.net domain, including `/demos/index.html`. I'm worry about the unintended consequences. Why don't we limit this to the offline mode assets, PHP builds, and WordPress builds?

Reply:
Why not cache these files? They are small and will load faster after refresh if they are cached.
We would also need to maintain a list similar to the one in `listAssetsRequiredForOfflineMode`.